### PR TITLE
fix(Core/Creature): Prevent flying creatures from falling during rebirth

### PR DESCRIPTION
**Pull request by pklloveyou**
## Changes Proposed:
-  ```.go c id 22441```
-  ```.die```
- ```.res all```
- Check its changes before and after

## Issues Addressed:

- Closes 

## Tests Performed:

- 
- 

## How to Test the Changes:

1.
2.
3.

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by Gultask** [comment]
Tested:

The Master's Eye (used to respawn on the ground before it was fixed through script)
Teribus (no waypoints, flying)
Torgos (with waypoints, flying)
A couple of stationary and patrolling creatures on the ground.

**Review by acidmanifesto — APPROVED** [review]

File: src/server/game/Entities/Unit/Unit.cpp [hunk 1]
```diff
@@ -19906,8 +19906,11 @@ void Unit::BuildMovementPacket(ByteBuffer* data) const
         *data << (float)m_movementInfo.splineElevation;
 }
 
-bool Unit::IsFalling() const
+bool Unit::IsFalling(bool spawn /*= false*/) const
 {
+    if (spawn)
+        return false;
+
```
Comment (acidmanifesto) [inline src/server/game/Entities/Unit/Unit.cpp:19913]:
```suggestion
    if (spawn)
   {
        return false;
    }
```

**Comment by acidmanifesto** [comment]
> Tested:
> 
> The Master's Eye (used to respawn on the ground before it was fixed through script) Teribus (no waypoints, flying) Torgos (with waypoints, flying) A couple of stationary and patrolling creatures on the ground.

Would you say this is good to go?

**Comment by UltraNix** [comment]
Nope - it's a hacky solution. I will fix it in other way.

**Pull request closed without merging**